demo/data-assimilation-4dvar/demo.ipynb

```diff
@@ -0,0 +1,855 @@
+    "\n",
+    "$$\n",
+    "\\mathbf{x}_i =\n",
+    "\\begin{cases}\n",
+    "\\mathbf{x}_0, & i = 0,\\\\\n",
+    "\\mathcal{M}(\\mathbf{x}_{i-1}), & i = 1,\\dots,n_{\\mathrm{obs}}-1.\n",
+    "\\end{cases}\n",
+    "$$\n",
+    "\n",
+    "**Definition of symbols**  \n",
+    "- $\\mathbf{x}_0\\in\\mathbb{R}^n$: the **analysis** (initial) state vector at the start of the window.  \n",
+    "- $\\mathbf{x}^b\\in\\mathbb{R}^n$: the **background** or first-guess state (short-range forecast).  \n",
+    "- $\\mathbf{B}\\in\\mathbb{R}^{n\\times n}$: background-error covariance matrix.  \n",
+    "- $n_{\\mathrm{obs}}$: number of observation times in the assimilation window.  \n",
+    "- $\\mathbf{y}_i\\in\\mathbb{R}^m$: observation vector at time $t_i$.  \n",
+    "- $\\mathbf{H}\\in\\mathbb{R}^{m\\times n}$: (linearized) observation operator mapping state $\\mathbf{x}_i$ to observation space.  \n",
+    "- $\\mathbf{R}\\in\\mathbb{R}^{m\\times m}$: observation-error covariance matrix.  \n",
+    "- $\\mathcal{M}:\\mathbb{R}^n\\to\\mathbb{R}^n$: nonlinear forecast model operator advancing the state from $t_{i-1}$ to $t_i$.  \n",
+    "- $\\mathbf{x}_i$: model state at time $t_i$, with $\\mathbf{x}_0$ as the control variable and $\\mathbf{x}_i = \\mathcal{M}(\\mathbf{x}_{i-1})$ thereafter.  \n",
+    "- $i$: index for observation times, ranging from 0 to $n_{\\mathrm{obs}}-1$.\n",
+    "\n",
+    "\n",
+    "#### What is being minimized?\n",
+    "\n",
+    "\n",
+    "1. **Observation‐misfit**\n",
+    "\n",
+    "   $$\n",
+    "   \\frac{1}{2}\n",
+    "   \\sum_{i=0}^{n_{\\mathrm{obs}}-1}\n",
+    "   \\bigl(\\mathbf{y}_i - \\mathbf{H}\\,\\mathbf{x}_i\\bigr)^\\top\n",
+    "   \\mathbf{R}^{-1}\n",
+    "   \\bigl(\\mathbf{y}_i - \\mathbf{H}\\,\\mathbf{x}_i\\bigr)\n",
+    "   $$\n",
+    "\n",
+    "   Enforces agreement with all observations, weighted by the observation‐error covariance $\\mathbf{R}$. However, real observations are always sparse and noisy, so minimizing this term alone is ill‐conditioned and admits infinitely many solutions.\n",
+    "\n",
+    "2. **Background‐misfit**  \n",
+    "   \n",
+    "   $$\n",
+    "   \\frac{1}{2}(\\mathbf{x}_0 - \\mathbf{x}^b)^\\top \\mathbf{B}^{-1} (\\mathbf{x}_0 - \\mathbf{x}^b)\n",
+    "   $$\n",
+    "\n",
+    "   Penalizes deviations from the imperfect forecast, scaled by the background‐error covariance $\\mathbf{B}$. This term also acts as a regularizer, making the overall optimization well‐posed.\n",
+    "\n",
+    "\n",
+    "#### Why not simply re-initialize the model with observations?\n",
+    "\n",
+    "There are many reasons -- here are two of them:\n",
+    "- **Noisy and sparse observations.** Direct insertion of $\\mathbf{y}$ injects noise and leaves unobserved components unconstrained.  \n",
+    "- **Temporal smoothing.** 4D-Var distributes corrections over the entire window, yielding a dynamically consistent analysis that honors both model physics and all observations.\n"
```

Comment:
To me, "honors" in this context makes me think "satisfy" -- but this is false. Maybe a wording like 
```suggestion
    "- **Temporal smoothing.** 4D-Var distributes corrections over the entire window, yielding a dynamically consistent analysis that tries to satisfy both model physics and all observations.\n"
```